### PR TITLE
Use secretref to provide bootstrap kubeconfig

### PR DESCRIPTION
**Pull request by fynluk**
**How to categorize this PR?**

/area usability
/area ops-productivity
/kind enhancement

**What this PR does / why we need it**:
This PR improved the HelmChart handling.

I created a new HelmChart value to avoid inline bootstrapkubeconfig. 
With this approach, you can now reference a kubeconfig for the bootstrap process instead of inline provisioning.

Use this new values to reference a secret: `values.yaml`
```yaml
global:
  gardenlet:
    gardenClientConnection:
      bootstrapKubeconfig:
        secretRef:
          name: <name of the secret>
          namespace: <namespace of the secret>
```

The install process of the gardenlet needs a bootstrap kubeconfig with a token. This token should be in general auto-generated, so I needed a HelmChart process to include this. I build a setup (outside of the gardenlet helm chart) to generate the token and also the bootstrap kubeconfig. Without this change, I have to provide this kubeconfig inline which is impossible with helm functions.
So that's the motivation of this PR

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by gardener-robot-ci-1** [comment]
Thank you @fynluk for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: charts/gardener/gardenlet/charts/runtime/templates/configmap-componentconfig.yaml [hunk 1]
```diff
@@ -32,8 +32,13 @@ data:
       {{- end }}
       {{- if .Values.global.gardenlet.config.gardenClientConnection.bootstrapKubeconfig }}
       bootstrapKubeconfig:
+        {{- if .Values.global.gardenlet.config.gardenClientConnection.bootstrapKubeconfig.secretRef }}
+        name: {{ required ".Values.global.gardenlet.config.gardenClientConnection.bootstrapKubeconfig.secretRef.name is required" .Values.global.gardenlet.config.gardenClientConnection.bootstrapKubeconfig.secretRef.name }}
+        namespace: {{ required ".Values.global.gardenlet.config.gardenClientConnection.bootstrapKubeconfig.secretRef.namespace is required" .Values.global.gardenlet.config.gardenClientConnection.bootstrapKubeconfig.secretRef.namespace }}
```
Comment (rfranzke) [inline charts/gardener/gardenlet/charts/runtime/templates/configmap-componentconfig.yaml:37]: `secretRef` is missing here?

File: charts/gardener/gardenlet/values.yaml [hunk 2]
```diff
@@ -41,6 +41,9 @@ global:
                              # then only this kubeconfig will be considered.
       #   name: gardenlet-kubeconfig-bootstrap
       #   namespace: garden
+      #   secretRef:
+      #     name: secretName of your bootstrapKubeconfig
+      #     namespace: namespace of your bootstrapKubeconfig
```
Comment (rfranzke) [inline charts/gardener/gardenlet/values.yaml]: Can you maybe add some comments here? I think it's confusing now that there is `name/namespace` and then `secretRef.name/secretRef.namespace`. A short sentence explaining the difference would shed some light :)

**Review by rfranzke — CHANGES_REQUESTED** [review]

**Comment by fynluk** [comment]
Sorry for my absence, I was on holiday. 

I added some documentation as you mentioned, but I didn't understand your first review. secretRef is given in lines 35 and 36. 
```yaml
        {{- if .Values.global.gardenlet.config.gardenClientConnection.bootstrapKubeconfig.secretRef }}
        name: {{ required ".Values.global.gardenlet.config.gardenClientConnection.bootstrapKubeconfig.secretRef.name is required" .Values.global.gardenlet.config.gardenClientConnection.bootstrapKubeconfig.secretRef.name }}
        namespace: {{ required ".Values.global.gardenlet.config.gardenClientConnection.bootstrapKubeconfig.secretRef.namespace is required" .Values.global.gardenlet.config.gardenClientConnection.bootstrapKubeconfig.secretRef.namespace }}
```
In the first line, I am checking if a secretRef is given. If so, the secret name and ns is stored in the configmap. If not, the bootstrapKubeconfig.name/ns fields are used

File: charts/gardener/gardenlet/charts/runtime/templates/configmap-componentconfig.yaml (hunk 1 above)
Comment (rfranzke) [reply charts/gardener/gardenlet/charts/runtime/templates/configmap-componentconfig.yaml:37]: Maybe I'm missing something, but shouldn't it look like this:

```yaml
      bootstrapKubeconfig:
        {{- if .Values.global.gardenlet.config.gardenClientConnection.bootstrapKubeconfig.secretRef }}
        secretRef:
          name: {{ required ".Values.global.gardenlet.config.gardenClientConnection.bootstrapKubeconfig.secretRef.name is required" .Values.global.gardenlet.config.gardenClientConnection.bootstrapKubeconfig.secretRef.name }}
          namespace: {{ required ".Values.global.gardenlet.config.gardenClientConnection.bootstrapKubeconfig.secretRef.namespace is required" .Values.global.gardenlet.config.gardenClientConnection.bootstrapKubeconfig.secretRef.namespace }}
        {{- else }}
```

**Review by rfranzke — CHANGES_REQUESTED** [review]

**Comment by fynluk** [comment]
In my understanding the componentconfig is bound as config for the gardenlet. And the bootstrapKubeconfig.secret/ns tells the gardenlet which secret he can use for the bootstrap process, isn't it?
So manipulating the componentconfig would trigger a code change.

**Comment by rfranzke** [comment]
You're right, of course... :) I missed this. Thanks!

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**